lib/puppet/face/certificate.rb

```diff
@@ -69,9 +69,9 @@
       # command line. If both specified, we'll fail.
       # jeffweiss 17 april 2012
 
-      global_setting_from_cli = Puppet.settings.set_by_cli?(:dns_alt_names) == true
-      raise ArgumentError, _("Can't specify both --dns_alt_names and --dns-alt-names") if options[:dns_alt_names] and global_setting_from_cli
-      options[:dns_alt_names] = Puppet[:dns_alt_names] if global_setting_from_cli
+      global_setting_from_cli = Puppet.settings.set_by_cli?(:subject_alt_names) == true
+      raise ArgumentError, _("Can't specify both --subject_alt_names and --subject-alt-names") if options[:subject_alt_names] and global_setting_from_cli
+      options[:subject_alt_names] = Puppet[:subject_alt_names] if global_setting_from_cli
```

Comment:
It's unclear if this bit here is necessary to keep around (and the above comment doesn't make it any better). We should think through what's actually going on here and clean this up (including making the comment better if we can).